### PR TITLE
Make app, group, and tag descriptions requirable

### PR DESCRIPTION
**Pull request by barborico**
# Overview of Changes

Descriptions are important for usability and auditability--let's let Access deployers make them required! This PR adds support for an environment variable `REQUIRE_DESCRIPTIONS` which, if set to `True`, will make descriptions required when creating/editing apps, groups, and tags.

## Front end environment loading

I made another change (the second commit) to how Vite loads environment so that it will pull variables set in `.env` files, but will still use the process env as the final source of truth. This was necessary to be able to test the frontend conditional form field requirements locally (or other config-based behavior like the web app name).

Before: Only read from shell/system environment
Now: Read from both .env files AND shell environment, with shell taking precedence

## Partial updates for apps, groups, and tags

To more consistently support Marshmallow's default loading without clearing properties omitted in a PUT request, I've updated the schema loading to apply partial updates.

# Testing

With `REQUIRE_DESCRIPTIONS=True`:
image

With `REQUIRE_DESCRIPTIONS=False` or omitting the variable altogether, the original behavior is preserved.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1211884278397459

### CODE REVIEW

File: api/views/schemas/core_schemas.py [hunk 1]
```diff
@@ -255,7 +255,7 @@ class OktaGroupSchema(SQLAlchemyAutoSchema):
             ),
         ),
     )
-    description = auto_field(load_default="", validate=validate.Length(max=1024))
```
Comment (somethingnew2-0) [inline api/views/schemas/core_schemas.py:258]: Just want to double-check, does this break serialization (dumping) when the database previously has empty descriptions? Do we need to add `dump_default=""`? https://marshmallow.readthedocs.io/en/stable/marshmallow.fields.html

Comment (eguerrant) [inline api/views/schemas/core_schemas.py:258]: Considering other companies use Access, I'm inclined to say that this should be put behind a feature flag/env variable. I imagine that other companies may not want to require descriptions and it could break their workflows to require them

Comment (barborico) [reply api/views/schemas/core_schemas.py:258]: The `description` columns in the DB aren't nullable, so I don't think it's an issue. But also no harm in adding that.

Claude's thoughts:
>  Even though the database columns have nullable=False and default="", adding dump_default="" ensures:
>  - Consistency: Matches the load_default="" behavior for symmetric serialization/deserialization
>  - Robustness: Handles any edge cases where the database might have NULL values (migrations, manual updates, etc.)
>  - Best Practice: Explicitly declares the serialization behavior rather than relying on implicit marshmallow defaults

Comment (barborico) [reply api/views/schemas/core_schemas.py:258]: Good call! Just added backwards compatibility and tested locally with `REQUIRE_DESCRIPTIONS` env var omitted (preserves existing behavior) or set to `True`.

File: api/config.py [hunk 2]
```diff
@@ -89,3 +89,6 @@ def default_user_search() -> list[str]:
 
 # Specify a custom app name
 APP_NAME = os.getenv("APP_NAME", "Access")
+
+# Require descriptions for apps, groups, and tags
+REQUIRE_DESCRIPTIONS = os.getenv("REQUIRE_DESCRIPTIONS", "False") == "True"
```
Comment (eguerrant) [inline api/config.py]: Feel free to leave as is but since the bool is being ingested as a string, would we want to be more flexible and allow things like 'true', 'True', 'TRUE' etc?

**Review by eguerrant — DISMISSED** [review]
If possible, please add tests for these changes

File: api/views/resources/app.py [hunk 3]
```diff
@@ -131,7 +131,7 @@ def put(self, app: App) -> ResponseReturnValue:
                 abort(400, "Only tags can be modified for the Access application")
 
         old_app_name = app.name
-        app = schema.load(request.json, instance=app)
+        app = schema.load(request.json, instance=app, partial=True)
```
Comment (somethingnew2-0) [inline api/views/resources/app.py:134]: Why was this and other examples in this PR updated to `partial=True`

Comment (barborico) [reply api/views/resources/app.py:134]: To more consistently support Marshmallow's default loading without clearing properties omitted in a PUT request. When I added `load_default=""` to the App description schema for consistency, it broke a unit test that was using partial updates which emptied the description. Hence, https://github.com/discord/access/pull/350/commits/c206b4c32e87ab26b615d16a0364887033ecc9da.

**Review by somethingnew2-0 — APPROVED** [review]

File: api/config.py (hunk 2 above)
Comment (somethingnew2-0) [reply api/config.py]: Yes, this and other boolean options in this file could be updated to 
```suggestion
REQUIRE_DESCRIPTIONS = os.getenv("REQUIRE_DESCRIPTIONS", "false").lower() == "true"
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -43,6 +43,8 @@ jobs:
         run: tox -e test
       - name: Test with tox with postgresql
         run: tox -e test-with-postgresql
+      - name: Test with tox with require descriptions
```
Comment (somethingnew2-0) [inline .github/workflows/ci.yml]: Instead of creating an entirely separate tox runner for a different edge-case we should probably use [pytest indirect parametrization](https://docs.pytest.org/en/latest/example/parametrize.html#indirect-parametrization) instead to set the environment variable on the `app()` fixture in https://github.com/discord/access/blob/main/tests/conftest.py#L34

It'd probably look something like
```
@pytest.fixture(scope="session")
def app(require_description: bool = False) -> Flask:
    load_dotenv(".testenv")
    if require_description:
       os.setenv("REQUIRE_DESCRIPTIONS", "True")
    app = create_app(testing=True)
    return app
 ```
 
 Then on some test like `test_put_app` in https://github.com/discord/access/blob/main/tests/test_app.py#L74
 ```
 @pytest.mark.parametrize("app", [False, True])
 def test_put_app(app: Flask,
    client: FlaskClient, db: SQLAlchemy, mocker: MockerFixture, access_app: App, app_group: AppGroup, tag: Tag
) -> None:
 ```

Comment (barborico) [reply .github/workflows/ci.yml]: Oooh, that looks a lot cleaner. Thanks for the tip!

**Review by somethingnew2-0 — APPROVED** [review]
